### ISSUE TITLE
Eggdrop tries to connect to port+1..3 even if port was open

### ISSUE THREAD

**Issue by michaelortmann** [eggheads/eggdrop#646]
Example:

Case A, no open port on the other side:
```
.link sslbot
[02:36:38] Linking to sslbot at 127.0.0.1:2514 ...
[02:36:38] net: attempted socket connection refused: 127.0.0.1
[02:36:38] net: attempted socket connection refused: 127.0.0.1
[02:36:38] net: attempted socket connection refused: 127.0.0.1
[02:36:38] net: attempted socket connection refused: 127.0.0.1
*** [linux] Couldn't link to sslbot.
```

Case B, open port on the other side, but its an ssl only port, and .link failed with connection refused:
```
.link sslbot
[02:36:53] Linking to sslbot at 127.0.0.1:2514 ...
[02:37:08] net: attempted socket connection refused: 127.0.0.1
[02:37:08] net: attempted socket connection refused: 127.0.0.1
[02:37:08] net: attempted socket connection refused: 127.0.0.1
*** [linux] Couldn't link to sslbot.
```

Notice, there is one debug line less. The one missing is for the active ssl port. See also #458.

We can distinguish those cases, and should not try those port+1..3 in case we found an open and responding port, which it seems, we just did.

**Comment by michaelortmann** [comment]
This issue is too hard to fix for 1.8 without too much of a hack / breaking compatibility. A bad attempt and a good starting point is hidden in the commits of #645. You can flag this issue 1.9+.

**Comment by vanosg** [comment]
OBE by #843
